src/interop.ts

```diff
@@ -36,23 +36,30 @@ export function firstToLower(str: string) {
  * @returns The normalized resource properties
  */
 export function normalize(value: any, cfnType?: string, pulumiProvider?: PulumiProvider): any {
+    return lift(value => normalizePromptValue(value, cfnType, pulumiProvider), value);
+}
+
+/**
+  * The inner part of `normalize` that does not have to deal with eventual types like Promise or pulumi.Output.
+  */
+function normalizePromptValue(value: any, cfnType?: string, pulumiProvider?: PulumiProvider): any {
     if (!value) return value;
 
     if (Array.isArray(value)) {
         const result: any[] = [];
         for (let i = 0; i < value.length; i++) {
-            result[i] = normalize(value[i], cfnType);
+            result[i] = normalizePromptValue(value[i], cfnType, pulumiProvider);
```

Comment:
This is a change but sounds like we need to pass pulumiProvider into the recurring calls? Was a bug perhaps?